### ISSUE TITLE
Missing NVTT.DLL with Pipeline tool and fresh VS2017

### ISSUE THREAD

**Issue by YugeshN** [MonoGame/MonoGame#5707]
I had an issue with my Win10 install and had to do a reset which cleaned out the OS and removed all the apps.

I re-installed VS2017 and Monogame. When trying to build my project, I am now getting the following error in the pipeline tool when processing an fbx:

Processor 'TextureProcessor' had unexpected failure!
System.DllNotFoundException: Unable to load dll 'nvtt': The specified module could not be found.

Any ideas what I need to install to get nvtt? I googled it, but can't seem to find which library / package nvtt comes with.

FYI, I am running build 3.7.0.347

Thanks.

**Comment by mrhelmut** [comment]
Did you forked the git submodules as well? NVTT is in the dependencies submodule and should be automatically copied to your output folder when building the pipeline.

**Comment by cra0zy** [comment]
It might be that some library that `nvtt` depends on is missing.

**Comment by YugeshN** [comment]
@mrhelmut I do see nvtt.dll in C:\Program Files (x86)\MSBuild\MonoGame\v3.0\Tools. So I am not sure why the pipeline tool is not finding it (and the pipeline binary appears to be in the same directory).

I think @cra0zy might be right. I don't think the error means that it can't find nvtt... I think it means that some module that nvtt depends on is missing.

Any suggestions on how I could find out what it's looking for?

**Comment by CanopyOverseer** [comment]
"Any suggestions on how I could find out what it's looking for?"

Use depends.exe > http://www.dependencywalker.com/

**Comment by KonajuGames** [comment]
Perhaps a MSVC runtime, such as from VS 2015 or VS 2013, depending on which one the library was built with.

**Comment by YugeshN** [comment]
@Birch-Grove Thanks for the tip. I did try depends.exe, and it's showing a whole slew of dependencies missing viz.


API-MS-WIN-CORE-APIQUERY-L1-1-0.DLL
API-MS-WIN-CORE-APPCOMPAT-L1-1-1.DLL
API-MS-WIN-CORE-APPINIT-L1-1-0.DLL
API-MS-WIN-CORE-ATOMS-L1-1-0.DLL
API-MS-WIN-CORE-COMM-L1-1-0.DLL
API-MS-WIN-CORE-CONSOLE-L1-1-0.DLL
API-MS-WIN-CORE-CONSOLE-L2-1-0.DLL
API-MS-WIN-CORE-CONSOLE-L3-1-0.DLL
API-MS-WIN-CORE-DATETIME-L1-1-1.DLL
API-MS-WIN-CORE-DATETIME-L1-1-2.DLL
API-MS-WIN-CORE-DEBUG-L1-1-0.DLL
API-MS-WIN-CORE-DEBUG-L1-1-1.DLL
API-MS-WIN-CORE-DELAYLOAD-L1-1-0.DLL
API-MS-WIN-CORE-DELAYLOAD-L1-1-1.DLL
API-MS-WIN-CORE-ENCLAVE-L1-1-0.DLL
API-MS-WIN-CORE-ERRORHANDLING-L1-1-0.DLL
API-MS-WIN-CORE-ERRORHANDLING-L1-1-1.DLL
API-MS-WIN-CORE-ERRORHANDLING-L1-1-3.DLL
API-MS-WIN-CORE-FIBERS-L1-1-1.DLL
API-MS-WIN-CORE-FIBERS-L2-1-1.DLL
API-MS-WIN-CORE-FILE-L1-1-0.DLL
API-MS-WIN-CORE-FILE-L1-2-1.DLL
API-MS-WIN-CORE-FILE-L1-2-2.DLL
API-MS-WIN-CORE-FILE-L2-1-1.DLL
API-MS-WIN-CORE-FILE-L2-1-2.DLL
API-MS-WIN-CORE-HANDLE-L1-1-0.DLL
API-MS-WIN-CORE-HEAP-L1-1-0.DLL
API-MS-WIN-CORE-HEAP-L1-2-0.DLL
API-MS-WIN-CORE-HEAP-L2-1-0.DLL
API-MS-WIN-CORE-HEAP-OBSOLETE-L1-1-0.DLL
API-MS-WIN-CORE-INTERLOCKED-L1-1-0.DLL
API-MS-WIN-CORE-IO-L1-1-0.DLL
API-MS-WIN-CORE-IO-L1-1-1.DLL
API-MS-WIN-CORE-JOB-L1-1-0.DLL
API-MS-WIN-CORE-KERNEL32-LEGACY-L1-1-1.DLL
API-MS-WIN-CORE-KERNEL32-PRIVATE-L1-1-1.DLL
API-MS-WIN-CORE-LARGEINTEGER-L1-1-0.DLL
API-MS-WIN-CORE-LIBRARYLOADER-L1-1-0.DLL
API-MS-WIN-CORE-LIBRARYLOADER-L1-2-0.DLL
API-MS-WIN-CORE-LIBRARYLOADER-L1-2-2.DLL
API-MS-WIN-CORE-LIBRARYLOADER-L2-1-0.DLL
API-MS-WIN-CORE-LOCALIZATION-L1-1-0.DLL
API-MS-WIN-CORE-LOCALIZATION-L1-2-1.DLL
API-MS-WIN-CORE-LOCALIZATION-L2-1-0.DLL
API-MS-WIN-CORE-LOCALIZATION-OBSOLETE-L1-3-0.DLL
API-MS-WIN-CORE-LOCALIZATION-PRIVATE-L1-1-0.DLL
API-MS-WIN-CORE-LOCALREGISTRY-L1-1-0.DLL
API-MS-WIN-CORE-MEMORY-L1-1-0.DLL
API-MS-WIN-CORE-MEMORY-L1-1-2.DLL
API-MS-WIN-CORE-MISC-L1-1-0.DLL
API-MS-WIN-CORE-NAMEDPIPE-L1-2-0.DLL
API-MS-WIN-CORE-NAMEDPIPE-L1-2-2.DLL
API-MS-WIN-CORE-NAMESPACE-L1-1-0.DLL
API-MS-WIN-CORE-NORMALIZATION-L1-1-0.DLL
API-MS-WIN-CORE-PRIVATEPROFILE-L1-1-1.DLL
API-MS-WIN-CORE-PROCESSENVIRONMENT-L1-1-0.DLL
API-MS-WIN-CORE-PROCESSENVIRONMENT-L1-2-0.DLL
API-MS-WIN-CORE-PROCESSSNAPSHOT-L1-1-0.DLL
API-MS-WIN-CORE-PROCESSTHREADS-L1-1-0.DLL
API-MS-WIN-CORE-PROCESSTHREADS-L1-1-2.DLL
API-MS-WIN-CORE-PROCESSTHREADS-L1-1-3.DLL
API-MS-WIN-CORE-PROCESSTOPOLOGY-L1-2-0.DLL
API-MS-WIN-CORE-PROFILE-L1-1-0.DLL
API-MS-WIN-CORE-PSAPI-ANSI-L1-1-0.DLL
API-MS-WIN-CORE-PSAPI-L1-1-0.DLL
API-MS-WIN-CORE-REALTIME-L1-1-0.DLL
API-MS-WIN-CORE-REGISTRY-L1-1-0.DLL
API-MS-WIN-CORE-RTLSUPPORT-L1-1-0.DLL
API-MS-WIN-CORE-RTLSUPPORT-L1-2-0.DLL
API-MS-WIN-CORE-SIDEBYSIDE-L1-1-0.DLL
API-MS-WIN-CORE-STRING-L1-1-0.DLL
API-MS-WIN-CORE-STRING-L2-1-0.DLL
API-MS-WIN-CORE-STRING-OBSOLETE-L1-1-0.DLL
API-MS-WIN-CORE-STRINGANSI-L1-1-0.DLL
API-MS-WIN-CORE-SYNCH-L1-1-0.DLL
API-MS-WIN-CORE-SYNCH-L1-2-0.DLL
API-MS-WIN-CORE-SYNCH-L1-2-1.DLL
API-MS-WIN-CORE-SYSINFO-L1-1-0.DLL
API-MS-WIN-CORE-SYSINFO-L1-2-1.DLL
API-MS-WIN-CORE-SYSINFO-L1-2-3.DLL
API-MS-WIN-CORE-SYSTEMTOPOLOGY-L1-1-0.DLL
API-MS-WIN-CORE-SYSTEMTOPOLOGY-L1-1-1.DLL
API-MS-WIN-CORE-THREADPOOL-LEGACY-L1-1-0.DLL
API-MS-WIN-CORE-THREADPOOL-PRIVATE-L1-1-0.DLL
API-MS-WIN-CORE-TIMEZONE-L1-1-0.DLL
API-MS-WIN-CORE-UTIL-L1-1-0.DLL
API-MS-WIN-CORE-WINDOWSERRORREPORTING-L1-1-0.DLL
API-MS-WIN-CORE-WINDOWSERRORREPORTING-L1-1-1.DLL
API-MS-WIN-CORE-WOW64-L1-1-0.DLL
API-MS-WIN-CORE-WOW64-L1-1-1.DLL
API-MS-WIN-CORE-XSTATE-L2-1-0.DLL
API-MS-WIN-CRT-PRIVATE-L1-1-0.DLL
API-MS-WIN-CRT-RUNTIME-L1-1-0.DLL
API-MS-WIN-CRT-STRING-L1-1-0.DLL
API-MS-WIN-CRT-TIME-L1-1-0.DLL
API-MS-WIN-EVENTING-PROVIDER-L1-1-0.DLL
API-MS-WIN-GDI-INTERNAL-UAP-L1-1-0.DLL
API-MS-WIN-SECURITY-APPCONTAINER-L1-1-0.DLL
API-MS-WIN-SECURITY-BASE-L1-2-0.DLL
MSVCP100.DLL
MSVCR100.DLL
API-MS-WIN-APPMODEL-RUNTIME-INTERNAL-L1-1-4.DLL
API-MS-WIN-APPMODEL-RUNTIME-L1-1-2.DLL
API-MS-WIN-CORE-COM-L1-1-1.DLL
API-MS-WIN-CORE-CRT-L1-1-0.DLL
API-MS-WIN-CORE-CRT-L2-1-0.DLL
API-MS-WIN-CORE-INTERLOCKED-L1-2-0.DLL
API-MS-WIN-CORE-REGISTRY-PRIVATE-L1-1-0.DLL
API-MS-WIN-CORE-THREADPOOL-L1-2-0.DLL
API-MS-WIN-CORE-VERSION-L1-1-0.DLL
API-MS-WIN-DEVICES-CONFIG-L1-1-1.DLL
API-MS-WIN-DEVICES-QUERY-L1-1-1.DLL
API-MS-WIN-EVENTING-CLASSICPROVIDER-L1-1-0.DLL
API-MS-WIN-EVENTING-OBSOLETE-L1-1-0.DLL
API-MS-WIN-POWER-BASE-L1-1-0.DLL
API-MS-WIN-POWER-SETTING-L1-1-0.DLL
API-MS-WIN-SECURITY-GROUPPOLICY-L1-1-0.DLL
API-MS-WIN-SECURITY-LSALOOKUP-L1-1-1.DLL
API-MS-WIN-SECURITY-LSAPOLICY-L1-1-0.DLL
API-MS-WIN-SECURITY-SDDL-L1-1-0.DLL
API-MS-WIN-SERVICE-MANAGEMENT-L1-1-0.DLL
API-MS-WIN-SERVICE-MANAGEMENT-L2-1-0.DLL
API-MS-WIN-SERVICE-PRIVATE-L1-1-1.DLL
API-MS-WIN-SERVICE-WINSVC-L1-2-0.DLL
EXT-MS-ONECORE-APPMODEL-STATEREPOSITORY-INTERNAL-L1-1-1.DLL
EXT-MS-WIN-ADVAPI32-NPUSERNAME-L1-1-0.DLL
EXT-MS-WIN-ADVAPI32-NTMARTA-L1-1-0.DLL
EXT-MS-WIN-ADVAPI32-REGISTRY-L1-1-0.DLL
EXT-MS-WIN-APPMODEL-DAXCORE-L1-1-1.DLL
EXT-MS-WIN-APPXDEPLOYMENTCLIENT-APPXDEPLOY-L1-1-0.DLL
EXT-MS-WIN-APPXDEPLOYMENTCLIENT-APPXDEPLOYONECORE-L1-1-0.DLL
EXT-MS-WIN-AUTHZ-CONTEXT-L1-1-0.DLL
EXT-MS-WIN-COM-PSMREGISTER-L1-2-2.DLL
EXT-MS-WIN-CORE-WINRT-REMOTE-L1-1-0.DLL
EXT-MS-WIN-EDPUTIL-POLICY-L1-1-1.DLL
EXT-MS-WIN-FECLIENT-ENCRYPTEDFILE-L1-1-1.DLL
EXT-MS-WIN-GDI-CLIPPING-L1-1-0.DLL
EXT-MS-WIN-GDI-DC-CREATE-L1-1-1.DLL
EXT-MS-WIN-GDI-DC-L1-2-0.DLL
EXT-MS-WIN-GDI-DC-L1-2-1.DLL
EXT-MS-WIN-GDI-DRAW-L1-1-1.DLL
EXT-MS-WIN-GDI-DRAW-L1-1-3.DLL
EXT-MS-WIN-GDI-FONT-L1-1-1.DLL
EXT-MS-WIN-GDI-FONT-L1-1-3.DLL
EXT-MS-WIN-GDI-INTERNAL-DESKTOP-L1-1-1.DLL
EXT-MS-WIN-GDI-METAFILE-L1-1-1.DLL
EXT-MS-WIN-GDI-METAFILE-L1-1-2.DLL
EXT-MS-WIN-GDI-PATH-L1-1-0.DLL
EXT-MS-WIN-GDI-PRINT-L1-1-0.DLL
EXT-MS-WIN-GDI-PRIVATE-L1-1-0.DLL
EXT-MS-WIN-GDI-WCS-L1-1-0.DLL
EXT-MS-WIN-GPAPI-GROUPPOLICY-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-APPCOMPAT-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-DATETIME-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-ERRORHANDLING-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-FILE-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-QUIRKS-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-QUIRKS-L1-1-1.DLL
EXT-MS-WIN-KERNEL32-REGISTRY-L1-1-0.DLL
EXT-MS-WIN-KERNEL32-SIDEBYSIDE-L1-1-0.DLL
EXT-MS-WIN-KERNELBASE-PROCESSTHREAD-L1-1-0.DLL
EXT-MS-WIN-MRMCORER-RESMANAGER-L1-1-0.DLL
EXT-MS-WIN-NTDSAPI-ACTIVEDIRECTORYCLIENT-L1-1-0.DLL
EXT-MS-WIN-NTDSAPI-ACTIVEDIRECTORYCLIENT-L1-1-1.DLL
EXT-MS-WIN-NTUSER-MESSAGE-L1-1-1.DLL
EXT-MS-WIN-NTUSER-STRING-L1-1-0.DLL
EXT-MS-WIN-NTUSER-SYNCH-L1-1-0.DLL
EXT-MS-WIN-NTUSER-WINDOW-L1-1-1.DLL
EXT-MS-WIN-PROFILE-EXTENDER-L1-1-0.DLL
EXT-MS-WIN-PROFILE-LOAD-L1-1-0.DLL
EXT-MS-WIN-RPC-SSL-L1-1-0.DLL
EXT-MS-WIN-RTCORE-NTUSER-SYSPARAMS-L1-1-0.DLL
EXT-MS-WIN-RTCORE-NTUSER-WINDOW-EXT-L1-1-0.DLL
EXT-MS-WIN-SECURITY-CAPAUTHZ-L1-1-1.DLL
EXT-MS-WIN-SECURITY-EFSWRT-L1-1-1.DLL
EXT-MS-WIN-SHELL32-SHELLCOM-L1-1-0.DLL

I suspect @KonajuGames is right, and one of the MSVC runtimes is missing... but it's not clear which one. It seems I already have the runtime for 2012, 2013 and 2017 installed. I can't seem to install 2015, since it complains I already have a newer runtime.

Maybe it needs 2015? If so, I am not sure how to get that installed if it sees I have something newer (2017?) and quits.

**Comment by KonajuGames** [comment]
It's the MSVCR100 and MSVCP100 libraries.  They are the runtimes for Visual C++ 2012.  See this forum post for links to the runtime installers.
http://answers.ea.com/t5/Technical-Issues/MSVCP100-dll-missing-in-windows-10/td-p/4702068

**Comment by Birch-Grove** [comment]
what @KonajuGames said. 

IIRC all those others are always showing that way due to the imports being set to delay-load

**Comment by Jjagg** [comment]
Related to #4485

**Comment by YugeshN** [comment]
So I thought it might somehow be related to me resetting Windows, so I did a completely clean install i.e. wiped the drive, and then install Windows 10 from scratch. Then installed VS2017, and tried building my project. Looks like the problem is still there.

FYI, before I did the clean install, I tried repairing all the VC runtimes... none of those fixed the problem.

**Comment by YugeshN** [comment]
So here's an interesting development... I installed Steam, Origin (EA Downloader), and a couple of the games I own on both Steam and Origin, I fired up the games... and then went back and rebuilt my project, and voila, it now works.

It looks like Steam, Origin, or one of the games installed through one of them also installed the necessary dependencies that nvtt.dll requires.

So my problem is solved for now... however, it would be great to know exactly what libraries / dlls the pipeline is dependent on and have that either statically linked (as per @Jjagg referenced) or bundled with the installer.

**Comment by tomspilman** [comment]
As @Jjagg mentioned... this is related to the missing VS C runtime.  See https://github.com/MonoGame/MonoGame/issues/4485 and https://github.com/MonoGame/MonoGame/issues/4263.

I'm closing this in favor of these other issues.

**Comment by cra0zy** [comment]
But @Jjagg is not the one who mentioned it :)

**Comment by YugeshN** [comment]
@tomspilman It's not clear if this problem has been fixed yet or not. 

I just did a fresh install of VS 2017 on a clean Windows 10 Professional install and ran into this problem again with Monogame 3.7.0.1163

I installed the missing VS C runtimes for both 2012 and 2013 (could not install the runtime for 2015 since it complains about a newer version being installed viz. 2017).

Installing the 2012 and 2013 runtimes did not fix the problem (I installed both the x86 and x64 runtimes).

The only way I could get around this issue was by using Steam and Origin and downloading and installing a bunch of games that somehow installed a runtime that resolves this.

It would be nice if we knew what the exact VS C runtime dependency was that will solve this issue.

FYI, if it helps, I previously did the same install on Windows 2010 Home, and it somehow worked after installing the 2012 and 2013 runtimes.

On my second clean install, I used Windows 10 Professional, and installing the 2012 and 2013 runtimes didn't fix this... so it seems the problem is slightly different (and more difficult to solve) on Windows 10 Pro than Windows 10 Home.
